python/tvm/ansor/auto_schedule.py

```diff
@@ -0,0 +1,206 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+User interface for Ansor auto-scheduler.
+
+The basic schedule search process for Ansor is designed to be:
+`Program sampling` -> `Performance Tuning`.
+
+In `Program sampling`, we use some predefined or heuristic rules to generate several initial
+schedules. Based on these initial start points, we have `Performance Tuning` to apply cost model
+and evolutionary search to seek for schedules with the best performance. Candidate schedules will
+be measured in the target hardware.
+"""
+
+import tvm._ffi
+from tvm.runtime import Object
+from .compute_dag import ComputeDAG
+from .measure import LocalBuilder, LocalRunner
+from . import _ffi_api
+
+
+@tvm._ffi.register_object("ansor.HardwareParams")
+class HardwareParams(Object):
+    """ The parameters of target hardware, this is used to guide the search process of
+    SearchPolicy.
+
+    TODO(...): This is considering to merge with the new Target:
+    https://discuss.tvm.ai/t/rfc-tvm-target-specification/6844
+
+    Parameters
```

Comment:
How do we decide these args? Some of them can be auto-detected from OS but what about ```max_unroll_vec``` and ```max_innermost_split_factor```?

Reply:
We can just set values like 32 or 64 for them. The search algorithm is not sensitive to them and the default values work well  for most platforms including Intel CPUs and ARM CPUs.

Reply:
The `max_unroll_vec` & `max_innermost_split_factor` does more likely to be user setting values rather than the hardware parameters.
Since they're not used in this minimum system, remove them for now, maybe we can add them to the parameter of SearchPolicy?

Reply:
I prefer we setting the default value rather than user specify it. The value is related with hardware more directly too (avx2 / avx512 / neon decides the max factor naturelly we should vectorize). The default value of 32 should work well on intel / arm.

Reply:
Yes, sometimes we may choose not to use all the cores available in the host system since some of the cores need to be kept for other purpose, such as data pre-processing, etc. So I think it's better to leave _num_cores_  a user-specified options with default value. 
Regarding to _vector_unit_bytes_/_cache_line_bytes_, in my understanding they are  primitive hardware configurations, maybe we can consider to auto-detect it? 
The principle is that we should do auto-detect as much as we can, without too much overkill.